CONTRIBUTING.md

```diff
@@ -155,6 +155,103 @@ toolkit for interacting with AWS backend resources.
 2. [AWS SDK for iOS](https://github.com/aws-amplify/aws-sdk-ios)
 3. [AWS SDK for JavaScript](https://github.com/aws/aws-sdk-js)
 
+## Integration Tests
+
+In addition to unit tests which mock Amplify API interaction, this repository has integration tests which 
+test functionality with real Amplify backends. The integration test script will execute tests in example 
+apps which have integration tests written (skipping those that don't). It runs on Android and iOS simulators.
+
+**Note:** To run integration tests, you will need prerequisite Amplify resources in the example 
+apps where the tests run. The process for creating those is noted below.
+
+To run all integration tests on available platforms:
+```console
+$ melos run test:integration
+```
+
+To run all tests just on Android (also works for `iPhone` instead of `android`):
+```console
+$ melos run test:integration:android
+```
+
+To run a single test file on device matching "sdk":
+```console
+$ cd packages/amplify_auth_cognito/example
+$ flutter drive --driver=test_driver/integration_test.dart --target=integration_test/sign_in_sign_out_test.dart -d sdk
+```
+
+## Provision Resources For Integration Tests
+
+Any app with integration tests will have a script `tool/provision_integration_test_resources.sh` which will call `amplify init` and `amplify push` with preconfigured amplify environments for that app. 
+Executing it will create real AWS resources. It requires [the Amplify CLI](https://docs.amplify.aws/cli). It does not need to run every time you run the tests. Run it once to set up or update your environments. 
+If you already have an amplify environment configured for an app, this command will create a "test" 
+environment and check it out.
+
+Create all the amplify environments in the example apps which have provisioning scripts (takes several minutes). Note that you may need to give yourself permission to execute the scripts.:
+```console
+$ melos run provision_integration_test_resources
+```
+
+The auth tests require some additional configuration to support lambda triggers for automatically 
+verifying temporary test users. Note that this should only be done for the test environment, never a production one. This can be done manually by [following this process](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-sign-up.html#aws-lambda-triggers-pre-registration-example-2) or by following these instructions for the amplify CLI:
+
+```console
```

Comment:
Do you find this ios example easier https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Auth/AWSCognitoAuthPluginIntegrationTests/README.md ? I could model it like that, but am otherwise open to ideas.

Reply:
I think I do find it a little easier to parse but it could be subjective

Reply:
ok tried to make it more like that